subprojects/jacoco/src/main/java/org/gradle/testing/jacoco/tasks/JacocoReportBase.java

```diff
@@ -181,7 +182,11 @@ public void executionData(TaskCollection tasks) {
      */
     @Internal
     public FileCollection getAllClassDirs() {
-        return classDirectories.plus(getAdditionalClassDirs());
+        FileCollection classes = classDirectories.plus(getAdditionalClassDirs());
+        JacocoTaskExtension jacocoTaskExtension = getProject().getTasks().named(JavaPlugin.TEST_TASK_NAME).get()
+            .getExtensions().getByType(JacocoTaskExtension.class);
+        return classes.getAsFileTree().matching(patterns ->
+            patterns.exclude(jacocoTaskExtension.getExcludes()).include(jacocoTaskExtension.getIncludes()));
```

Comment:
I think this needs to be configured at the plugin level, not here.

Reply:
Your goal is to configure what the convention is. But users should be able to override the convention. They way you've done it here, you lock in the convention.

Reply:
I think the place you want to add this filter is to somewhere in `JacocoPlugin::addDefaultReportTask`.